app/main.py

```diff
@@ -1,25 +1,103 @@
+
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Slide, name: str) -> None:
```

Comment:
```suggestion
    def __set_name__(self, owner: object, name: str) -> None:
```
It's better to use object, for example you can validate another class's data, it's a universal class. The same in other methods.